ag_data/tests/test_error_log.py

```diff
@@ -0,0 +1,50 @@
+from django.test import TestCase
+import datetime
+
+from ag_data.models import ErrorLog
+
+
+class TestErrorLogModels(TestCase):
+    test_error_log_data = {
+        "missing_error": {
+            "error_timestamp": datetime.datetime(2020, 4, 17, 13, 17, 00),
+            "error_code": "MISSING_COLUMN",
+            "error_description": "Seneor_id Column Missing",
+            "error_raw_data": r'{"date":"2020-04-15 02:25:13.880456","values": {"temperature":"35.56126022338867"}}',
+        },
+        "invalid_error": {
+            "error_timestamp": datetime.datetime(2020, 4, 17, 14, 24, 00),
+            "error_code": "INVALID_COLUMN_VALUE",
+            "error_description": "Sensor_id 100 Invalid",
+            "error_raw_data": r'{date":"2020-03-01 02:25:13.112234","sensor_id":"100",\
+                                    "values": {"temperature":"35.56126022338867"}}',
+        },
+    }
+
+    def setUp(self):
+        for error in self.test_error_log_data.keys():
+            ErrorLog.objects.create(
+                error_timestamp=self.test_error_log_data[error]["error_timestamp"],
+                error_code=self.test_error_log_data[error]["error_code"],
+                error_description=self.test_error_log_data[error]["error_description"],
+                error_raw_data=self.test_error_log_data[error]["error_raw_data"],
+            )
+
+    def test_error_log_table(self):
```

Comment:
Could we test all possible error code choices? To check that we have all error code as we expected, with correct strings/names. One idea would be to use the dictionary, and a for-loop.

Reply:
I will add more test cases in the test data, actually, I am already using dictionary in test cases.

Reply:
surely! what I meant was, to have a dictionary to include all error code choices, and use it in the test.